proto/scenario.proto

```diff
@@ -0,0 +1,116 @@
+syntax = "proto3";
+package proto;
+
+message Scenario {
+  Load load = 1;
+  Cluster cluster = 2;
+  Workload workload = 3;
+  repeated Plugin plugin = 4;
+}
+
+message Load {
+  oneof arrival_rate_oneof {
+    LoadConstant constant = 1;
+    LoadPeriodic periodic = 3;
+    LoadCustom custom = 3;
+  }
+  oneof request_weight_oneof {
+    RequestConstant constant = 4;
+  }
+}
+
+message LoadConstant {
+  int32 arrivals_per_second = 1;
+}
+
+message LoadPeriodic {
+  oneof shape_oneof {
+    ShapeSine sine = 1;
+    ShapeAsdr asdr = 2;
+  }
+}
+
+message ShapeSine {
+  LoadInterval interval = 1;
+}
+
+message ShapeAsdr {
+  LoadInterval initial = 1;
+  LoadInterval attack = 2;
+  LoadInterval sustain = 3;
+  LoadInterval decay = 4;
+  LoadInterval release = 5;
+}
+
+message LoadCustom {
+  repeated LoadInterval interval = 1;
+}
+
+message LoadInterval {
+  int32 width_milliseconds = 1;
+  int32 amplitude_aps = 2;
+}
+
+message RequestConstant {
+  int32 cpu_milliseconds = 1;
+  int32 io_milliseconds = 2;
+  int32 memory_bytes = 3;
+}
+
+message Cluster {
+  // infinite capacity
+}
+
+message Workload {
+  int32 readiness_delay_milliseconds = 1;
```

Comment:
Will this be applied uniformly to every copy of a workload? Or for each instance? At the moment Skenario assumes fixed, identical startup times but I'd always wanted to replace that with something more realistic down the road.

Reply:
Yes.

I would like to leave room for a statistical readiness delay model.  So perhaps I should put this in a struct.

Reply:
+1 to struct.